definitions/features/upstream_repositories.rb

```diff
@@ -0,0 +1,24 @@
+class Features::UpstreamRepositories < ForemanMaintain::Feature
+  metadata do
+    label :upstream_repositories
+    description 'Feature for operations on upstream repositories'
+  end
+
+  def repoids_and_urls
+    repoids_and_urls = {}
+    repository_manager.enabled_repos_hash.each do |repo, url|
+      repo_urls.each do |regex|
+        repoids_and_urls[repo] = url if url =~ regex
+      end
+    end
+    repoids_and_urls
+  end
+
+  private
+
+  def repo_urls
+    [%r{yum.theforeman.org\/*},
```

Comment:
I think `yum.theforeman.org` is enough to know if repo is upstream? Instead of trying to check separate paths of Foreman, Katello, plugins, pulpcore etc?

Reply:
yeah, for *current* upstream releases, this is always on `yum.theforeman.org` , different subfolders, but that shouldn't matter for your matcher